### PR TITLE
Upgrade typescript tools version on mobile-app

### PR DESCRIPTION
**Pull request by SteveKennedy**
My first commit updated TypeScript tools version. This will avoid Visual Studio Update 1 & Update 2 developers from getting the dialog alerting them to having a older version of type script tools in this project.  The dialog happens automatically and by default. Only the solution with with the mobile-app project in it will be effected.

### CODE REVIEW

**Comment by SteveKennedy** [comment]
I imagine that users of Visual Studio Update 1 & 2 would get alerted about this when they first open the solution with the mobile-app. Visual Studio 2015 Update 1 included 1.6 and 1.7 releases of the TypeScript tools. Visual Studio 2015 Update 2 includes TypeScript 1.8.  The latest version (1.8) can be downloaded separately at https://www.microsoft.com/en-us/download/details.aspx?id=48593. These typescript tools updates brings significant enhancements to TypeScript's type system and enables support for the polymorphic type, intersection types, local type declarations, generic type aliasing, and user-defined type guard functions. It also completes ES6 support in TypeScript by adding ES6 Generators and ES6 Class expressions, and brings support for new ES7 feature proposals like ES7 Exponentiation operator and ES7 Async functions. The compiler now highlights common bugs such as unreachable code, missing return statements, and unused labels.

File: AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Controllers/SiteAdminControllerShould.cs [hunk 1]
```diff
@@ -628,6 +641,6 @@ public void ControllerHasAuthorizeAtttributeWithTheCorrectPolicy()
         private static Mock<UserManager<ApplicationUser>> CreateApplicationUserMock()
         {
             return new Mock<UserManager<ApplicationUser>>(Mock.Of<IUserStore<ApplicationUser>>(), null, null, null, null, null, null, null, null, null);
-        }    
+        }
```
Comment (SteveKennedy) [inline AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Controllers/SiteAdminControllerShould.cs]: For some stupid reason, I removed the trailing spaces.

File: AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Controllers/SiteAdminControllerShould.cs [hunk 2]
```diff
@@ -136,9 +136,22 @@ public void EditUserGetSendsUserByUserIdQueryWithCorrectUserId()
...
+
+            string userId = It.IsAny<string>();
+            mediator.Setup(x => x.Send(It.Is<UserByUserIdQuery>(q => q.UserId == userId)))
+                .Returns(new ApplicationUser());
+            var controller = new SiteController(null, logger.Object, mediator.Object);
+            
+            var result = controller.EditUser(userId);
+            var model = ((ViewResult)result).ViewData.Model as EditUserModel;
+                        
+            Assert.Equal(model.Organization, null);
```
Comment (SteveKennedy) [inline AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Controllers/SiteAdminControllerShould.cs]: Can I assume model cannot be null at this point?

File: AllReadyApp/Mobile-App/AllReadyApp.jsproj [hunk 3]
```diff
@@ -78,7 +78,7 @@
     <VisualStudioVersion>14.0</VisualStudioVersion>
   </PropertyGroup>
   <PropertyGroup>
-    <TypeScriptToolsVersion>1.5</TypeScriptToolsVersion>
+    <TypeScriptToolsVersion>1.8</TypeScriptToolsVersion>
```
Comment (SteveKennedy) [inline AllReadyApp/Mobile-App/AllReadyApp.jsproj]: Background: Visual Studio 2015 Update 1 included 1.6 and 1.7 releases of the TypeScript tools. Visual Studio 2015 Update 2 includes TypeScript 1.8.  The latest version (1.8) can be downloaded separately at https://www.microsoft.com/en-us/download/details.aspx?id=48593. These typescript tools updates brings significant enhancements to TypeScript's type system and enables support for the polymorphic type, intersection types, local type declarations, generic type aliasing, and user-defined type guard functions. It also completes ES6 support in TypeScript by adding ES6 Generators and ES6 Class expressions, and brings support for new ES7 feature proposals like ES7 Exponentiation operator and ES7 Async functions. The compiler now highlights common bugs such as unreachable code, missing return statements, and unused labels.

**Comment by tonysurma** [comment]
Thanks @SteveKennedy for your contributions!

Generally yes, separate 'small' Pull requests are best so we can review and accept separately.

If you need reference for how to have more than one pull request at a time I believe one of @stevejgordon @dpaquette and @MisterJames have documented that (cc @BillWagner )

If you separate or pull back the second part of this PR we can review and accept each PR.

Thanks!

**Comment by SteveKennedy** [comment]
I don't know what I'm doing.  I tried reverting the second commit.  Ultimately, only the first commit changes are now in this PR.  There's some line break/spacing issue in the SiteAdminControllerShould.cs that I couldn't seem to revert.

File: AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Controllers/SiteAdminControllerShould.cs [hunk 4]
```diff
@@ -630,4 +630,4 @@ private static Mock<UserManager<ApplicationUser>> CreateApplicationUserMock()
             return new Mock<UserManager<ApplicationUser>>(Mock.Of<IUserStore<ApplicationUser>>(), null, null, null, null, null, null, null, null, null);
         }    
     }
-}
```
Comment (SteveKennedy) [inline AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Controllers/SiteAdminControllerShould.cs]: I have no idea how to revert this change.

**Comment by SteveKennedy** [comment]
I closed this pull request and will open a new one with the (cleaner) changes.

**Comment by dpaquette** [comment]
Hi @SteveKennedy. You might find this tutorial helpful. 

http://www.davepaquette.com/archive/2016/01/24/Submitting-Your-First-Pull-request.aspx

I put it together based on my own struggles when I was getting started with GitHub and Pull Requests. If you have any questions, feel free to reach out. We can always do a screenshare and work through issues together.
